stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/reactions/ViewReactionsView.kt

```diff
@@ -0,0 +1,132 @@
+package io.getstream.chat.android.ui.messages.reactions
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.graphics.Path
+import android.util.AttributeSet
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.utils.extensions.dpToPx
+import io.getstream.chat.android.ui.utils.extensions.getColorCompat
+
+public class ViewReactionsView : ReactionsView {
+
+    private val theirsMessageReactionsBubblePaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        style = Paint.Style.FILL
+        color = context.getColorCompat(R.color.stream_grey_90)
+    }
+
+    private val myMessageReactionsBubblePaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        style = Paint.Style.FILL
+        color = Color.WHITE
+    }
+
+    private val myMessageReactionsBubbleStrokePaint = Paint(Paint.ANTI_ALIAS_FLAG).apply {
+        // one dp stroke looks too thin
+        strokeWidth = 1.dpToPx() * 1.5f
+        style = Paint.Style.STROKE
+        color = context.getColorCompat(R.color.stream_grey_90)
+    }
+
+    public constructor(context: Context) : super(context) {
+        init(context, null)
+    }
+
+    public constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        init(context, attrs)
+    }
+
+    public constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+        context,
+        attrs,
+        defStyleAttr
+    ) {
+        init(context, attrs)
+    }
+
+    private fun init(context: Context, attrs: AttributeSet?) {
+        setStyle(ViewReactionsViewStyle(context, attrs))
+    }
+
+    override fun createReactionItems(message: Message, isMyMessage: Boolean): List<ReactionsAdapter.ReactionItem> {
+        val reactionsMap = mutableMapOf<String, ReactionsAdapter.ReactionItem>()
+        message.latestReactions.forEach { reaction ->
+            val ownReaction = message.ownReactions.any { it.type == reaction.type }
+            val alreadyPresent = reactionsMap.containsKey(reaction.type)
+            if (!alreadyPresent || ownReaction) {
+                reactionsMap[reaction.type] = ReactionsAdapter.ReactionItem(reaction, ownReaction)
+            }
+        }
+        return reactionsMap.values
+            .toList()
+            .sortedBy { !it.isMine }
+    }
+
+    override fun setMessage(message: Message, isMyMessage: Boolean) {
+        super.setMessage(message, isMyMessage)
+        val horizontalPadding = if (message.latestReactions.size == 1) {
+            0
+        } else {
+            reactionsViewStyle.reactionsHorizontalPadding
+        }
+        setPadding(horizontalPadding, 0, horizontalPadding, 0)
+    }
+
+    override fun drawReactionsBubble(canvas: Canvas, isMyMessage: Boolean) {
```

Comment:
Seeing some odd things with this View being drawn. For own messages, the outer edges of the bubbles occasionally get clipped, and the large and small trailing bubbles are connected in an odd way as well.

![image](https://user-images.githubusercontent.com/12054216/100597436-c1889300-32fd-11eb-9017-10caa9c49445.png)

Reply:
Nice catch, looks like android doesn't provide a way to draw border inside path.

Reply:
<img width="500" alt="Screenshot 2020-12-01 at 11 14 10" src="https://user-images.githubusercontent.com/9600921/100714751-4095ce00-33c7-11eb-8efd-04f8cdce9d1e.png">

Reply:
Added half stroke offset, looks better

Reply:
![image](https://user-images.githubusercontent.com/12054216/100720997-13452200-33bf-11eb-80cd-10b6260d0799.png)

This one specifically still seems off, both the cropping and the small line connecting the circles. Might not be worth the trouble now, we can get back to polishing it later on.

Reply:
Later on :)